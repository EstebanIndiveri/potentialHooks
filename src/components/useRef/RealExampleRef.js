import React, { useState } from 'react';
import '../02-useEffect/SimpleForm.css';
import MultipleCustomHooks from '../examples/MultipleCustomHooks'
const RealExampleRef = () => {

    const[show,setShow]=useState(false);

  

    return ( <div>
        <h1>Real Example</h1>
            <hr/>
            {
                show&&<MultipleCustomHooks/>
            }
            <button className="btn btn-warning mt-5"
            onClick={()=>{
                setShow(!show);
            }}
            >
                show/hide
            </button>
        </div> );
}
 
export default RealExampleRef;