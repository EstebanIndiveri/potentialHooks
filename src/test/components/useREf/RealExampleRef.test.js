import React from 'react';
import {shallow} from 'enzyme';
import '@testing-library/jest-dom';
import RealExampleRef from '../../../components/useRef/RealExampleRef'

describe('pruebas en <RealExampleRef/>', () => {
    const wrapper=shallow(<RealExampleRef/>)
    test('muestra correctamente', () => {
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(false);

    });
    test('muestra el componente <MultipleCustomHooks/>', () => {
        wrapper.find('button').simulate('click');
        expect(wrapper.find('MultipleCustomHooks').exists()).toBe(true);

    });
})
