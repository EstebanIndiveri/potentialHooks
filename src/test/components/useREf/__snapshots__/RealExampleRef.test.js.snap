// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en <RealExampleRef/> muestra correctamente 1`] = `
<div>
  <h1>
    Real Example
  </h1>
  <hr />
  <button
    className="btn btn-warning mt-5"
    onClick={[Function]}
  >
    show/hide
  </button>
</div>
`;
